File: plugin/zeppelin/src/main/scala/org/intellij/plugin/zeppelin/models/Exceptions.scala
package org.intellij.plugin.zeppelin.models


class ZeppelinException() extends Exception {
  override def getMessage: String = {
    s"Error during Zeppelin Exception."
  }
}

case class ZeppelinConnectionException(uri: String) extends ZeppelinException {
  override def getMessage: String = {
    s"Cannot connect to the Zeppelin app. " +
      s"Check the availability of web socket connection to the service $uri"
  }
}

case class ZeppelinLoginException() extends ZeppelinException {
  override def getMessage: String = {
    s"Cannot login to the Zeppelin app. " +
      s"The login or the password is wrong."
  }
}


case class NotebookNotFoundException(id: String) extends ZeppelinException {
  override def getMessage: String = {
    s"Notebook with $id id is not found."
  }
}

File: zeppelin/src/org/intellij/plugin/zeppelin/models/Exceptions.kt
package org.intellij.plugin.zeppelin.models

open class ZeppelinException(val msg:String?=null) : Exception() {
    override val message: String
        get() = "Error in Zeppelin plugin. ${msg?:""}"
}

data class ZeppelinConnectionException(val uri: String) : ZeppelinException() {
    override val message: String
        get() = "Cannot connect to the Zeppelin app. Check the availability of web socket connection to the service $uri"
}

class ZeppelinLoginException : ZeppelinException() {
    override val message: String
        get() = "Cannot login to the Zeppelin app. The login or the password is wrong."
}

data class NotebookNotFoundException(val id: String) : ZeppelinException() {
    override val message: String
        get() = "Notebook with $id id is not found."
}

data class ParagraphNotFoundException(val executeContext: ExecuteContext) : ZeppelinException() {
    override val message: String
        get() = "Paragraph with is not found. Data: $executeContext"
}

class SessionIsClosedException : ZeppelinException()

class ParseException(private val json: String,
                     private val parseClass: String) : ZeppelinException() {
    override val message: String
        get() = "Cannot parse json to object.\nJson:\n$json\nClass:\n$parseClass"

}